forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,250 @@
+        # It records the spans replaced by new texts.
+        # It is a map from datapack id to a list of tuples
+        # (span, new text) inserted by :func:`replace`.
+        # The new text will be used for building new data pack.
+        self._replaced_annos: DefaultDict[
+            int, SortedList[Tuple[Span, str]]
+        ] = defaultdict(lambda: SortedList([], key=lambda x: x[0]))
+
+        # :attr:`_inserted_annos_pos_len`: {datapack id: Dict{position: length}}
+        # It records the position and length of inserted spans,
+        # mapping from datapack id to a dictionary (position -> length)
+        # inserted by :func:`insert`.
+        # The position is the index in the original datapack
+        # of insertion, and the length is the length of the inserted string.
+        self._inserted_annos_pos_len: DefaultDict[
+            int, SortedDict[int, int]
+        ] = defaultdict(lambda: SortedDict({}))
+
+        # :attr:`_deleted_annos_id`: {datapack id: Set[annotation tid]}
+        # It records the deleted span ids, mapping from datapack id
+        # to a set of annotation tids appended by :func:`delete`.
+        self._deleted_annos_id: DefaultDict[int, Set[int]] = defaultdict(set)
+
+        # :attr:`_data_pack_map`: {orig pack id: new pack id}
+        # It maintains a mapping from the pack id
+        # of the original pack to the pack id of augmented pack.
+        # It is used when copying the MultiPackLink and MultiPackGroup.
+        self._data_pack_map: Dict[int, int] = {}
+
+        # :attr:`_entry_maps`: {datapack id: Dict{orig tid, new tid}}
+        # It is a map for tracking the annotation ids
+        # before and after the auto align. It maps the
+        # original annotation tid to the new annotation tid.
+        # It is used when copying the Link/Group/MultiPackLink/MultiPackGroup.
+        self._entry_maps: Dict[int, Dict[int, int]] = {}
+
+        self.configs = configs
+
+    def _overlap_with_existing(self, pid: int, begin: int, end: int) -> bool:
+        r"""
+        This function will check whether the new span
+        has an overlap with any existing spans.
+
+        Args:
+            pid: Datapack Id.
+            begin: The span begin index.
+            end: The span end index.
+        Returns:
+            True if the input span overlaps with
+            any existing spans, False otherwise.
+        """
+        if len(self._replaced_annos[pid]) == 0:
+            return False
+        ind: int = (
+            bisect_left(self._replaced_annos[pid], (Span(begin, begin), "")) - 1
+        )
+        if ind < 0:
+            ind += 1
+
+        while ind < len(self._replaced_annos[pid]):
+            span: Span = self._replaced_annos[pid][ind][0]
+            if not (span.begin >= end or span.end <= begin):
+                return True
+            if span.begin > end:
+                break
+            ind += 1
+
+        return False
+
+    def inserted_annotation_status(self):
```

Comment:
let's annotate the return type on the function signature

Reply:
If the purpose is to return one of the internal value, consider using `@property`?

Reply:
Btw, is it necessary or good to expose internal states to users? We developers can still access these internal variables in test cases without this function